From 4aac93962a6fcd3eb8e043976735ac8799ded7d7 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sat, 2 Aug 2014 20:56:21 +0200
Subject: [PATCH 1005/1005] spdif: cubox-i/hummingboard: Use only PLL4
 (internal) as clock source

The spdif driver will enumerate all possible clock sources (internal
and external) for a matching frequency. Since CuBox-i does not have
any external oscillator, we must tell the driver so. Otherwise it may
choose a non-present clock source resulting in no sound at 88.2 and
176.4kHz.

Signed-off-by: Rudi <r.ihle@s-t.de>
---
 arch/arm/boot/dts/imx6dl-hummingboard.dts |   10 ++++++++++
 arch/arm/boot/dts/imx6qdl-cubox-i.dtsi    |   10 ++++++++++
 2 files changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/imx6dl-hummingboard.dts b/arch/arm/boot/dts/imx6dl-hummingboard.dts
index ad39709..998e1f2 100644
--- a/arch/arm/boot/dts/imx6qdl-hummingboard.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-hummingboard.dtsi
@@ -233,6 +233,16 @@
 };
 
 &spdif {
+	clocks = <&clks 197>, <&clks 0>,
+		  <&clks 197>, <&clks 0>,
+		  <&clks 0>,   <&clks 0>,
+		  <&clks 0>,  <&clks 0>,
+		  <&clks 0>, <&clks 156>;
+	clock-names = "core",  "rxtx0",
+		      "rxtx1", "rxtx2",
+		      "rxtx3", "rxtx4",
+		      "rxtx5", "rxtx6",
+		      "rxtx7", "dma";
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi
index d238c7e4..fbcfe44 100644
--- a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi
@@ -248,6 +248,16 @@
 };
 
 &spdif {
+	clocks = <&clks 197>, <&clks 0>,
+		  <&clks 197>, <&clks 0>,
+		  <&clks 0>,   <&clks 0>,
+		  <&clks 0>,  <&clks 0>,
+		  <&clks 0>, <&clks 156>;
+	clock-names = "core",  "rxtx0",
+		      "rxtx1", "rxtx2",
+		      "rxtx3", "rxtx4",
+		      "rxtx5", "rxtx6",
+		      "rxtx7", "dma";
 	status = "okay";
 };
 
-- 
1.7.9.5

